wfe2/verify.go

```diff
@@ -54,31 +56,40 @@ func sigAlgorithmForKey(key *jose.JSONWebKey) (jose.SignatureAlgorithm, error) {
 	return "", errors.New("JWK contains unsupported key type (expected RSA, or ECDSA P-256, P-384, or P-521)")
 }
 
-var supportedAlgs = map[string]bool{
-	string(jose.RS256): true,
-	string(jose.ES256): true,
-	string(jose.ES384): true,
-	string(jose.ES512): true,
+var supportedAlgs = map[jose.SignatureAlgorithm]bool{
+	jose.RS256: true,
+	jose.ES256: true,
+	jose.ES384: true,
+	jose.ES512: true,
+}
+
+// getSupportedAlgs returns a sorted slice of joseSignatureAlgorithm's from a
+// map of boulder allowed signature algorithms.
+func getSupportedAlgs() []jose.SignatureAlgorithm {
+	sigAlgs := maps.Keys(supportedAlgs)
+	slices.Sort(sigAlgs)
```

Comment:
Rather than calling `maps.Keys` each time (given this is called on each verify), we should change `supportedAlgs` to be an array so we can pass a slice of that array directly to ParseSigned.